File: tests/fixtures/discrimination/go/out_of_corpus/counter.go
package main

import "fmt"

var count int

func increment() {
	count = count + 1
}

func decrement() {
	count = count - 1
}

func reset() {
	count = 0
}

func getCount() int {
	return count
}

func main() {
	increment()
	increment()
	increment()
	decrement()
	fmt.Println(getCount())
}

File: tests/fixtures/discrimination/rust/out_of_corpus/counter.rs
struct Counter {
    value: i32,
}

impl Counter {
    fn new() -> Counter {
        Counter { value: 0 }
    }

    fn increment(&mut self) {
        self.value = self.value + 1;
    }

    fn decrement(&mut self) {
        self.value = self.value - 1;
    }

    fn get(&self) -> i32 {
        self.value
    }

    fn reset(&mut self) {
        self.value = 0;
    }
}

fn main() {
    let mut c = Counter::new();
    c.increment();
    c.increment();
    c.decrement();
    println!("Count: {}", c.get());
}
